src/common/snippets/src/op/subgraph.cpp

```diff
@@ -504,16 +507,18 @@ snippets::Schedule snippets::op::Subgraph::generate(ngraph::pass::Manager& opt,
     // actual code emission
     ngraph::snippets::code ptr = m_generator->generate(m_body, generatorConfig, compile_params);
 
-    // check that body doesn't have constants for scheduling
+    // check that body doesn't have constants or non-decomposed ops for scheduling
     std::vector<std::shared_ptr<opset1::Constant>> constants;
     for (auto op : m_body->get_ordered_ops()) {
-        if (auto constant = ov::as_type_ptr<opset1::Constant>(op)) {
-            if (ngraph::shape_size(constant->get_shape()) != 1 && constant->get_shape() != Shape()) {
-                constants.push_back(constant);
-            }
+        if ((ov::is_type<opset1::Constant>(op) && ov::shape_size(op->get_shape()) != 1 && op->get_shape() != Shape()) ||
+             ov::is_type<ov::op::v1::Softmax>(op) ||
+             ov::is_type<ov::op::v8::Softmax>(op) ||
+             ov::is_type<ov::op::v1::Transpose>(op) ||
+             ov::is_type<ov::op::v1::Broadcast>(op) ||
+             ov::is_type<ov::op::v1::Reshape>(op)) {
+            throw ngraph::ngraph_error("External op detected: " + std::string(op->get_type_name()) + ". Snippet is illigal for scheduling");
```

Comment:
Don't like this check at all. What if a plugin decide to add custom transformation that will insert such operations into the body. 
I expect there should be exception inside m_generator->generate(...) since no corresponding emitter provided. Isn't it enough?